### PR TITLE
Feat: Cancel streams using permissions

### LINKED ISSUE #2215: Streaming Payments API: Cancelling streams using permissions
Parent issue: https://github.com/JoinColony/colonyCDapp/issues/2209

- New saga needs to be implemented
- Relevant block-ingestor handlers need to be implemented
- A corresponding colony action should be created
- It should be possible to annotate the transaction

### PR DESCRIPTION
**Pull request by iamsamgibbs**
[See block-ingestor counterpart](https://github.com/JoinColony/block-ingestor/pull/217)

## Description

Users should be able to cancel streaming payments.

This PR implements a new saga and a block-ingestor handler.

## Testing

1. Install the streaming payments extension
2. Create a new streaming payment using the temporary UI on the dashboard
3. Run this query and note the end time of the streaming payment

```
query MyQuery {
  listStreamingPayments {
    items {
      endTime
      startTime
      nativeId
    }
  }
}
```

4. Using the temporary dashboard UI, cancel the streaming payment (ids start at 1)
5. Run the query again and notice the endTime change
6. Verify a new colony action was created by running this query. The latest action should have a streamingPaymentId.

```
query MyQuery {
  listColonyActions {
    items {
      streamingPaymentId
      id
      streamingPayment {
        id
        endTime
        createdAt
      }
    }
  }
}
```

## Diffs

**New stuff** ✨

* Cancel streaming payment saga

**Changes** 🏗

* Colony action now accepts streaming payment ID

Resolves #2215

### CODE REVIEW

**Review by ArmandoGraterol — APPROVED** [review]
LGTM! Replicated the instructions and I was able to check that the action is being created:
Screenshot 2024-05-06 at 15 39 14

File: amplify/backend/backend-config.json
Comment (jakubcolony) [inline amplify/backend/backend-config.json:1]: I don't think this should be committed

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Great start on this PR @iamsamgibbs! Everything seems to be working well, including annotations:
image

The only problem I spotted is that messages seem to be missing for cancel transactions:
image

I have 2 more requests which have not been specifically covered in the issue but would appreciate if they could be added:
 - Link between `StreamingPayment` and `ColonyActions` model, ie. `actions` field on the streaming payment that would contain all actions related to the stream (currently just `CANCEL_STREAMING_PAYMENT`)
 - A field on the `StreamingPayment` model to indicate that the stream was cancelled - I think a boolean flag will suffice for now

**Comment by iamsamgibbs** [comment]
Made those changes / additions as requested @jakubcolony

**Review by jakubcolony — APPROVED** [review]
LGTM, thanks for making the changes 👍 

image

**Review by rdig — APPROVED** [review]
Nicely done! :100:

**Pull request merged**